api/service.go

```diff
@@ -621,15 +777,22 @@ func (s *service) AddFile(server pb.API_AddFileServer) error {
 	pth, err := s.ipfsClient.Unixfs().Add(
 		server.Context(),
 		ipfsfiles.NewReaderFile(reader),
-		options.Unixfs.Pin(true),
+		options.Unixfs.Pin(false),
 		options.Unixfs.Progress(true),
 		options.Unixfs.Events(eventCh))
 	if err != nil {
 		return err
 	}
 
-	_, err = s.collections.Files.Create(server.Context(), pth, name, proj.ID)
-	if err != nil && !strings.Contains(err.Error(), store.ErrUniqueExists.Error()) {
+	dirpth, err := s.ipfsClient.Object().
+		AddLink(server.Context(), folderPath, fileName, pth, options.Object.Create(true))
+	if err != nil {
+		return err
+	}
+	if err = s.ipfsClient.Pin().Update(server.Context(), folderPath, dirpth); err != nil {
```

Comment:
Move the pin to the root.

Reply:
Is this because adding a link changed the CID, so the original one should be unpinned because has _stalled_ data?

Reply:
Yep!